third_party/move/move-compiler-v2/src/env_pipeline/lambda_lifter.rs

```diff
@@ -732,6 +767,15 @@ impl<'a> ExpRewriterFunctions for LambdaLifter<'a> {
         let body = ExpRewriter::new(env, &mut replacer).rewrite_exp(body.clone());
         let fun_id = FunId::new(fun_name);
         let params_types = params.iter().map(|param| param.get_type()).collect();
+        if abilities.has_store() {
+            let loc = env.get_node_loc(id);
+            env.error(
+                &loc,
+                // TODO(LAMBDA)
+                "Lambdas expressions with `store` ability currently may only be a simple call to an existing `public` function.  This lambda expression requires defining a `public` helper function, which might affect module upgradeability and is not yet supported."
```

Comment:
This error message looks a little bit confusing for users (e.g what is a `helper` function and how it may affect upgradeability`). Maybe just remove `requires defining a `public` helper function, which might affect module upgradeability` and just saying it is not supported yet?

Reply:
done.